datadog_checks_dev/datadog_checks/dev/tooling/commands/meta/snmp/generate_profile.py

```diff
@@ -0,0 +1,661 @@
+    ::= { entPhysicalContainsTable 1 }
+    ```
+    or its json dump, where `INDEX` is replaced by indices
+    ```json
+    "entPhysicalContainsEntry": {
+        "name": "entPhysicalContainsEntry",
+        "oid": "1.3.6.1.2.1.47.1.3.3.1",
+        "nodetype": "row",
+        "class": "objecttype",
+        "maxaccess": "not-accessible",
+        "indices": [
+          {
+            "module": "ENTITY-MIB",
+            "object": "entPhysicalIndex",
+            "implied": 0
+          },
+          {
+            "module": "ENTITY-MIB",
+            "object": "entPhysicalChildIndex",
+            "implied": 0
+          }
+        ],
+        "status": "current",
+        "description": "A single container/'containee' relationship."
+      },
+    ```
+    Sometimes indexes are columns from another table, and we might want to use another column as it could have more
+    human readable information - we might prefer to see the interface name vs its numerical table index. This can be
+    achieved using metric_tag_aliases
+
+
+    Return a list of SNMP metrics and copy its yaml dump to the clipboard
+    Metric tags need to be added manually
+    """
+    # create a list of all mib files directories and mib names
+    source_directories = set()
+    mibs = set()
+    for file in mib_files:
+        source_directories.add(os.path.dirname(file))
+        mibs.add(os.path.splitext(os.path.basename(file))[0])
+    # create a tmp dir for compiled json mibs
+    json_destination_directory = os.path.join(gettempdir(), 'mibs')
+    if not os.path.exists(json_destination_directory):
+        os.mkdir(json_destination_directory)
+
+    profile_oid_collection = {}
+    # build profile
+    for oid_node in _extract_oids_from_mibs(
+        list(mibs), list(source_directories), json_destination_directory, filter_path_oids
+    ):
+        if oid_node.node_type == 'table':
+            _add_profile_table_node(profile_oid_collection, oid_node)
+        elif oid_node.node_type == 'row':
+            # requires
+            _add_profile_row_node(
+                profile_oid_collection,
+                oid_node,
+                os.path.dirname(mib_files[0]),
+                metric_tag_aliases_path=aliases_metric_tag_path,
+                json_mib_directory=json_destination_directory,
+            )
+        elif oid_node.node_type == 'column':
+            _add_profile_column_node(profile_oid_collection, oid_node)
+        elif oid_node.node_type == 'scalar':
+            _add_profile_scalar_node(profile_oid_collection, oid_node)
+
+    pyperclip.copy(yaml.dump({'metrics': list(profile_oid_collection.values())}, sort_keys=False))
+    print('📋 Profile yaml dump copied to your clipboard')
+
+
+@click.command(context_settings=CONTEXT_SETTINGS, short_help='Join two SNMP profiles including unique nodes')
+@click.argument('profile_a')
+@click.argument('profile_b')
+@click.pass_context
+def join_profiles(ctx, profile_a, profile_b):
```

Comment:
I used it to merge an existing profile with one generated from the tool, to add only missing nodes. It needs more work and documentation, so I didn't expose it to `ddev`, I will remove it as I see it can be misleading